src/include/migraphx/reflect.hpp

```diff
@@ -111,8 +111,17 @@ auto reflect(T& x, Selector f)
 template <class T>
 auto reflect_tie(T& x)
 {
-    return reflect(x, [](auto&& y, auto&&...) { return detail::wrap<decltype(y)>(y); })(
-        [](auto&&... xs) { return detail::auto_tuple(xs.get()...); });
+    return reflect(x, [](auto&& y, auto&&...) {
+        if constexpr(is_reflectable<decltype(y)>{}) // cppcheck-suppress UnnecessaryElseStatement
```

Comment:
Does removing the else not compile?

Also I think you can put the `cppcheck-suppress UnnecessaryElseStatement` on the line above.

Reply:
yes, If i remove `else` it doesn't compile.

Reply:
Moved `cppcheck-suppress` to line above.